lib/executor/ramping_arrival_rate_test.go

```diff
@@ -243,6 +243,109 @@ func TestRampingArrivalRateRunCorrectRateWithSlowRate(t *testing.T) {
 	assert.Equal(t, int64(2), es.GetInitializedVUsCount())
 }
 
+func TestRampingArrivalRateRunGracefulStop(t *testing.T) {
+	t.Parallel()
+	et, err := lib.NewExecutionTuple(nil, nil)
+	require.NoError(t, err)
+	es := lib.NewExecutionState(lib.Options{}, et, 10, 10)
+
+	runner := simpleRunner(func(ctx context.Context) error {
+		time.Sleep(5 * time.Second)
+		return nil
+	})
+	ctx, cancel, executor, _ := setupExecutor(
+		t, &RampingArrivalRateConfig{
+			TimeUnit: types.NullDurationFrom(1 * time.Second),
+			Stages: []Stage{
+				{
+					Duration: types.NullDurationFrom(2 * time.Second),
+					Target:   null.IntFrom(10),
+				},
+			},
+			StartRate:       null.IntFrom(10),
+			PreAllocatedVUs: null.IntFrom(10),
+			MaxVUs:          null.IntFrom(10),
+			BaseConfig: BaseConfig{
+				GracefulStop: types.NullDurationFrom(15 * time.Second),
```

Comment:
Nit: you could reduce this duration and the sleep to speed the test up, there's no reason for it to take 5s+. Even though it's not noticeable when running the full suite, it matters when running it standalone.